lib/core/client.js

```diff
@@ -506,7 +477,7 @@ class Parser extends HTTPParser {
     // have no way of slicing the parsing buffer without knowing
     // the offset which is only provided in kOnExecute.
     if (upgrade) {
-      const head = this.getCurrentBuffer().slice(ret)
```

Comment:
Maybe we could implement this using the buffer view in `parser.execute`?

Reply:
I'm fine with it as is for now...